### ISSUE TITLE
Automatically decode html entities

### ISSUE THREAD

**Issue by arcanis** [howdyai/botkit#240]
Shouldn't messages received from Slack be automatically html-decoded ?

**Comment by jlsjonas** [comment]
I agree, I now have to replace &amp; to & (and in principle the same with &lt; & &gt; ) for url to properly parse the url; alternatively provide a clean() method (in order to decode only when required)

**Comment by benbrown** [comment]
@jlsjonas thanks for the input -- did you see this when using the Event API or the RTM connection? Or both?

**Comment by jlsjonas** [comment]
If I remember correctly the project never used the RTM connection; so event API. Both might be affected though (untested)

**Comment by benbrown** [comment]
OK yes, I can confirm that the entities come through encoded with the events api.

Now the question is, should this happen by default, or via an optional middleware?

**Comment by benbrown** [comment]
This little snippet of code will do the job, but since it requires a new dependency that would require extra work to keep up to date, I'm not sure if we should include this in core.  THOUGHTS?

```
const Entities = require('html-entities').AllHtmlEntities; 
const entities = new Entities();
// Decode HTML-encoded entities in Slack messages
controller.middleware.receive.use((bot, message, next) => {
  if (bot.type === 'slack' && message.text) {
     message.text = entities.decode(message.text);
  }
  next();  
});
```

**Comment by benbrown** [comment]
Makes sense. Thanks for the recommendation on a better entity package.
